@here/harp-mapview-decoder/lib/TileDataSource.ts

```diff
@@ -249,25 +253,23 @@ export class TileDataSource<TileType extends Tile> extends DataSource {
         if (tileLoader === undefined) {
             return;
         }
-        if (tileLoader.decodedTile !== undefined) {
-            this.setDecodedTileOnTile(tileLoader.decodedTile, tile);
-        } else {
-            tileLoader
-                .loadAndDecode()
-                .then(tileLoaderState => {
-                    assert(tileLoaderState === TileLoaderState.Ready);
-                    const decodedTile = tileLoader.decodedTile;
-                    this.setDecodedTileOnTile(decodedTile, tile);
-                })
-                .catch(tileLoaderState => {
-                    if (
-                        tileLoaderState !== TileLoaderState.Canceled &&
-                        tileLoaderState !== TileLoaderState.Failed
-                    ) {
-                        this.logger.error("Unknown error" + tileLoaderState);
-                    }
-                });
-        }
+
+        tileLoader
```

Comment:
this changes the implementation of the method, should `updateTile` be renamed?

Reply:
I'm not sure why you think the name needs to be changed, now the tile is always updated if you call the method.
Maybe you mean it should better be `loadTile()`?
I've realized there's no reason to go through TileDataSource to do what we're doing in here. I think we should move this method to `Tile.load()`.
The same with `setDecodedTileOnTile()`, which can be moved to `Tile.setDecodedTile()` `decodedTileHasGeometry()`, which should be moved to `DecodedTile.hasGeometry()`.

Reply:
`loadTile` to me gives a clearer picture of what is happening behind the scenes, `update` to me seems like it implies there is data that is added and is less precise (but I understand than an update can of course be implemented in terms of a complete re loading of the data).
The changes you made to not go through the `TileDataSource` also make it much cleaner!

Reply:
at least the performance implications of `loadTile` are a bit clearer also than `updateTile`